### PR TITLE
Defined get_lldp_neighbors_detail() method for JunOS, NXOS, EOS & IOS-XR

### PR DESCRIPTION
**Pull request by mirceaulinic**
Child #154 
Although get_lldp_neighbors() does exist, there is a need to retrieve further details of the LLDP neighbors such as capabilities, system description etc.
We preferred to define a different method in order to not overlap with existing work. In the future they might be merged into a single method.

### CODE REVIEW

**Comment by dbarrosop** [comment]
More conflicts...

Regarding having two methods. I think it's better to have multiple methods. Otherwise the code becomes entangled and testing becomes harder.

File: napalm/base.py [hunk 1]
```diff
@@ -364,3 +364,41 @@ def get_interfaces_counters(self):
...
         """
         raise NotImplementedError
+
+    def get_lldp_neighbors_detail(self, interface = ''):
+        """
+        Returns a detailed view of the LLDP neighbors as a dictionary
+        containing lists of dictionaries for each interface.
+
+        Inner dictionaries contain fields:
+            * parent_interface (string)
```
Comment (dbarrosop) [inline napalm/base.py]: What is this attribute?

**Comment by jejenone** [comment]
rebased, squashed and cleaned

File: napalm/base.py (hunk 1 above)
Comment (mirceaulinic) [inline napalm/base.py]: parent_interface? The name of the aggregated interface

**Comment by mirceaulinic** [comment]
I think this one is ready too

File: napalm/base.py (hunk 1 above)
Comment (jejenone) [inline napalm/base.py]: @dbarrosop 
```
jerome@re0.edge01.sjc01> show lldp neighbors
Local Interface    Parent Interface    Chassis Id          Port info          System Name
xe-0/1/1           ae7                 00:1c:73:ee:c0:46   Ethernet3/29/1     core01.sjc01
xe-0/3/3           ae7                 00:1c:73:ee:c0:46   Ethernet3/29/2     core01.sjc01
```

**Pull request merged**